### ISSUE TITLE
Issue with failing DNS lookups

### ISSUE THREAD

**Issue by JamesHarrison** [rubinius/rubinius#2303]
```
rubinius 2.0.0.rc1 (1.9.3 dec3ff31 yyyy-mm-dd JI) [x86_64-unknown-linux-gnu]
Linux fleeting-glance 3.2.0-40-generic #64-Ubuntu SMP Mon Mar 25 21:22:10 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux
```

When making requests with Resolv.getaddress for DNS entries for domains that do not exist - as is fairly common in things like DNSBL tools - a NoMethodError is thrown by resolv as opposed to returning nil/false.

```
rbx-head :015 > require 'resolv'
 => true
rbx-head :015 > Resolv.getaddress("spamhaus.org")
 => "157.56.178.135"
rbx-head :016 > Resolv.getaddress("sbl.spamhaus.org")
NoMethodError: undefined method `ip_address_list' on Socket (Class)
        from kernel/delta/kernel.rb:81:in `ip_address_list (method_missing)'
        from /usr/local/rvm/rubies/rbx-head/lib/19/resolv.rb:399:in `use_ipv6?'
        from /usr/local/rvm/rubies/rbx-head/lib/19/resolv.rb:392:in `each_address'
        from /usr/local/rvm/rubies/rbx-head/lib/19/resolv.rb:115:in `each_address'
        from kernel/bootstrap/array.rb:68:in `each'
        from /usr/local/rvm/rubies/rbx-head/lib/19/resolv.rb:114:in `each_address'
        from /usr/local/rvm/rubies/rbx-head/lib/19/resolv.rb:92:in `getaddress'
        from /usr/local/rvm/rubies/rbx-head/lib/19/resolv.rb:43:in `getaddress'
        from (irb):16
        from kernel/common/block_environment.rb:75:in `call_on_instance'
        from kernel/common/eval.rb:75:in `eval'
        from kernel/common/kernel19.rb:42:in `loop'
        from kernel/bootstrap/proc.rb:22:in `call'
        from kernel/common/throw_catch19.rb:8:in `catch'
        from kernel/common/throw_catch.rb:10:in `register'
        from kernel/common/throw_catch19.rb:7:in `catch'
        from kernel/bootstrap/proc.rb:22:in `call'
        from kernel/common/throw_catch19.rb:8:in `catch'
        from kernel/common/throw_catch.rb:10:in `register'
        from kernel/common/throw_catch19.rb:7:in `catch'
        from /usr/local/rvm/gems/rbx-head/gems/railties-3.2.13/lib/rails/commands/console.rb:47:in `start'
        from /usr/local/rvm/gems/rbx-head/gems/railties-3.2.13/lib/rails/commands/console.rb:8:in `start'
        from /usr/local/rvm/gems/rbx-head/gems/railties-3.2.13/lib/rails/commands.rb:41:in `__script__'
        from kernel/common/codeloader.rb:212:in `require'
        from kernel/common/kernel.rb:638:in `require'
        from script/rails:6:in `__script__'
        from kernel/delta/codeloader.rb:68:in `load_script'
        from kernel/delta/codeloader.rb:118:in `load_script'
        from kernel/loader.rb:616:in `script'
        from ker
```

**Comment by cmc476** [comment]
bump...only thing keeping me from using rbx right now for a production app.

**Comment by zuk** [comment]
I ran into this with the Datastax Cassandra client. Stacktrace looked like this:

```
NoMethodError: undefined method `ip_address_list' on Socket (Class)
kernel/delta/kernel.rb:78:in `ip_address_list (method_missing)'
rbx-2.5.1/gems/gems/rubysl-resolv-2.1.0/lib/rubysl/resolv/resolv.rb:415:in `use_ipv6?'
rbx-2.5.1/gems/gems/rubysl-resolv-2.1.0/lib/rubysl/resolv/resolv.rb:408:in `each_address'
rbx-2.5.1/gems/gems/rubysl-resolv-2.1.0/lib/rubysl/resolv/resolv.rb:115:in `each_address'
kernel/bootstrap/array.rb:76:in `each'
rbx-2.5.1/gems/gems/rubysl-resolv-2.1.0/lib/rubysl/resolv/resolv.rb:114:in `each_address'
rbx-2.5.1/gems/gems/rubysl-resolv-2.1.0/lib/rubysl/resolv/resolv.rb:57:in `each_address'
rbx-2.5.1/gems/gems/cassandra-driver-2.1.3/lib/cassandra.rb:486:in `cluster_async'
```

Workaround was to add an ugly monkey patch like this:

if RUBY_ENGINE == 'rbx'
  class Resolv
    class DNS
      def use_ipv6?
        return false
      end
    end
  end
end

**Comment by YorickPeterse** [comment]
Fixed by the recent release of rubysl-socket, see https://github.com/rubysl/rubysl-socket for more information. These changes are available as of Rubinius 3.3.
